core/corehttp/gateway_handler.go

```diff
@@ -264,38 +269,30 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 	}
 }
 
-func (i *gatewayHandler) postHandler(w http.ResponseWriter, r *http.Request) {
-	nd, err := i.newDagFromReader(r.Body)
-	if err != nil {
-		internalWebError(w, err)
-		return
-	}
-
-	k, err := i.node.DAG.Add(nd)
-	if err != nil {
-		internalWebError(w, err)
-		return
+func (i *gatewayHandler) postHandler(ctx context.Context, w http.ResponseWriter, r *http.Request) {
+	// TODO(cryptix): can we move this into ServeHTTP, so we can do it once for all methods?
```

Comment:
Okay, I want to do this together with tests in another PR. I think this is big enough as it is.